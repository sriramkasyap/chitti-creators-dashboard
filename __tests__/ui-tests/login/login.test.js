import router from "next/router";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Login from "../../../pages/login";
import AuthProvider, { AuthContext } from "../../../contexts/AuthContext";

describe("Test Login initial render", () => {
  it("should match the snapshot", () => {
    const container = render(
      <AuthProvider>
        <Login />
      </AuthProvider>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("Test Login form", () => {
  it("should render the basic fields", () => {
    const { getByTestId } = render(
      <AuthProvider>
        <Login />
      </AuthProvider>
    );
    expect(getByTestId("email")).toBeInTheDocument();
    expect(getByTestId("password")).toBeInTheDocument();
    expect(getByTestId("login-button")).toBeInTheDocument();
  });

  it("should check input values and submit form", () => {
    const contextValue = {
      loggedInUser: { profile: { fullName: "", displayPicture: null } },
      loginError: "",
      userLogin: jest.fn(),
      fetchMe: jest.fn(),
    };

    const { getByTestId } = render(
      <AuthContext.Provider value={contextValue} router={router}>
        <Login />
      </AuthContext.Provider>
    );

    fireEvent.change(getByTestId("email"), {
      target: { value: "john@doe.com" },
    });
    expect(getByTestId("email").value).toBe("john@doe.com");

    fireEvent.change(getByTestId("password"), { target: { value: "123456" } });
    expect(getByTestId("password").value).toBe("123456");

    fireEvent.click(getByTestId("login-button"));

    expect(contextValue.userLogin).toHaveBeenCalledTimes(1);
  });

  it("should check empty input value", () => {
    const contextValue = {
      loggedInUser: { profile: { fullName: "", displayPicture: null } },
      loginError: "",
      userLogin: jest.fn(),
      fetchMe: jest.fn(),
    };

    const { getByTestId, getAllByText } = render(
      <AuthContext.Provider value={contextValue} router={router}>
        <Login />
      </AuthContext.Provider>
    );

    fireEvent.change(getByTestId("email"), {
      target: { value: "" },
    });

    fireEvent.change(getByTestId("password"), { target: { value: "" } });

    fireEvent.click(getByTestId("login-button"));

    expect(getAllByText("This Field is Required!")).toHaveLength(2);
    expect(contextValue.userLogin).not.toBeCalled();
  });

  it("should check email validation", () => {
    const { getByTestId, getByText, queryByText } = render(
      <AuthProvider>
        <Login />
      </AuthProvider>
    );

    fireEvent.change(getByTestId("email"), {
      target: { value: "john@doe" },
    });

    fireEvent.click(getByTestId("login-button"));
    expect(getByText("Invalid Email")).toBeInTheDocument();

    fireEvent.change(getByTestId("email"), {
      target: { value: "john" },
    });

    fireEvent.click(getByTestId("login-button"));
    expect(getByText("Invalid Email")).toBeInTheDocument();

    fireEvent.change(getByTestId("email"), {
      target: { value: "john@doe.com" },
    });

    fireEvent.click(getByTestId("login-button"));
    expect(queryByText("Invalid Email")).not.toBeInTheDocument();
  });
});
