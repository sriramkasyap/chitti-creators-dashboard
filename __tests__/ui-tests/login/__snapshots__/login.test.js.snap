// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login initial render should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1wymqtv"
      >
        <div
          class="css-u8sorr"
        >
          <div
            class="css-1z0snx"
          >
            <p
              class="chakra-text css-0"
            >
              Creating content
            </p>
            <p
              class="chakra-text css-0"
            >
              has never been more
            </p>
            <p
              class="chakra-text css-0"
            >
              rewarding
            </p>
          </div>
        </div>
        <div
          class="css-qz51m7"
        >
          <div
            class="css-11qvk8e"
          >
            <p
              class="chakra-text css-0"
            >
              Creating content
            </p>
            <p
              class="chakra-text css-0"
            >
              has never been more
            </p>
            <p
              class="chakra-text css-0"
            >
              rewarding
            </p>
          </div>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-1wq58g"
          />
          <div
            class="css-zdqkti"
          >
            <p
              class="chakra-text css-14yvhw0"
            >
              Login to
            </p>
            <h2
              class="chakra-heading css-1bwgkrq"
            >
              Chitti.
            </h2>
            <div
              class="chakra-form-control css-1gifr5s"
              role="group"
            >
              <label
                class="chakra-form__label css-b5d9cx"
                for="email"
                id="email-label"
              >
                Email address
              </label>
              <input
                class="chakra-input css-1bp9x2g"
                data-testid="email"
                id="email"
                name="emailId"
                type="email"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-r07iet"
              role="group"
            >
              <label
                class="chakra-form__label css-b5d9cx"
                for="password"
                id="password-label"
              >
                Password
              </label>
              <input
                class="chakra-input css-1bp9x2g"
                data-testid="password"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <button
              class="chakra-button css-1ccbrzu"
              data-testid="login-button"
              type="button"
            >
              Login
            </button>
          </div>
          <div
            class="css-1bnxh4l"
          >
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-cal9d6"
            />
            <p
              class="chakra-text css-h1djnr"
            >
              or
            </p>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-cal9d6"
            />
          </div>
          <div
            class="css-u6v8er"
          >
            <p
              class="chakra-text css-2od7nb"
            >
              Don’t have an account yet?
            </p>
            <a
              class="chakra-link css-9rwuyw"
              href="/signup"
            >
              <p
                class="chakra-text css-1xdtch0"
              >
                Sign up
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1wymqtv"
    >
      <div
        class="css-u8sorr"
      >
        <div
          class="css-1z0snx"
        >
          <p
            class="chakra-text css-0"
          >
            Creating content
          </p>
          <p
            class="chakra-text css-0"
          >
            has never been more
          </p>
          <p
            class="chakra-text css-0"
          >
            rewarding
          </p>
        </div>
      </div>
      <div
        class="css-qz51m7"
      >
        <div
          class="css-11qvk8e"
        >
          <p
            class="chakra-text css-0"
          >
            Creating content
          </p>
          <p
            class="chakra-text css-0"
          >
            has never been more
          </p>
          <p
            class="chakra-text css-0"
          >
            rewarding
          </p>
        </div>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-1wq58g"
        />
        <div
          class="css-zdqkti"
        >
          <p
            class="chakra-text css-14yvhw0"
          >
            Login to
          </p>
          <h2
            class="chakra-heading css-1bwgkrq"
          >
            Chitti.
          </h2>
          <div
            class="chakra-form-control css-1gifr5s"
            role="group"
          >
            <label
              class="chakra-form__label css-b5d9cx"
              for="email"
              id="email-label"
            >
              Email address
            </label>
            <input
              class="chakra-input css-1bp9x2g"
              data-testid="email"
              id="email"
              name="emailId"
              type="email"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-r07iet"
            role="group"
          >
            <label
              class="chakra-form__label css-b5d9cx"
              for="password"
              id="password-label"
            >
              Password
            </label>
            <input
              class="chakra-input css-1bp9x2g"
              data-testid="password"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="chakra-button css-1ccbrzu"
            data-testid="login-button"
            type="button"
          >
            Login
          </button>
        </div>
        <div
          class="css-1bnxh4l"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-cal9d6"
          />
          <p
            class="chakra-text css-h1djnr"
          >
            or
          </p>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-cal9d6"
          />
        </div>
        <div
          class="css-u6v8er"
        >
          <p
            class="chakra-text css-2od7nb"
          >
            Don’t have an account yet?
          </p>
          <a
            class="chakra-link css-9rwuyw"
            href="/signup"
          >
            <p
              class="chakra-text css-1xdtch0"
            >
              Sign up
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
